prdoc/pr_5961.prdoc

```diff
@@ -0,0 +1,15 @@
+# Schema: Polkadot SDK PRDoc Schema (prdoc) v1.0.0
+# See doc at https://raw.githubusercontent.com/paritytech/polkadot-sdk/master/prdoc/schema_user.json
+
+title: "Bounties Pallet: add `approve_bounty_with_curator` call"
+
+doc:
+  - audience: [Runtime Dev, Runtime User]
+    description: |
+      Adds `approve_bounty_with_curator` call to the bounties pallet to combine `approve_bounty` and `propose_curator` into one call. If `unassign_curator` is called after `approve_bounty_with_curator` the process falls back to the previous flow of calling `propose_curator` separately.
```

Comment:
I think we also need to mention new status variant, and how it behaves. 
Ideally we would document the status machine, right now clients has to read through the code base to understand it. It could be a doc for `BountyStatus` type.